lib/transaction/response.ex

```diff
@@ -0,0 +1,70 @@
+defmodule Xander.Transaction.Response do
+  @moduledoc """
+  Parses the response from a transaction. Accepts CBOR encoded responses.
+  """
+
+  require Logger
+
+  @doc """
+  Parses the response from a transaction. Accepts CBOR encoded responses.
+
+  ## Examples
+
+      # Transaction accepted
+      iex> binary = Xander.Util.plex_encode(CBOR.encode([1]))
+      iex> Xander.Transaction.Response.parse_response(binary)
+      {:ok, :accepted}
+
+      # Transaction rejected with reason
+      iex> tag = %CBOR.Tag{tag: :bytes, value: "invalid tx"}
+      iex> binary = Xander.Util.plex_encode(CBOR.encode([2, tag]))
+      iex> Xander.Transaction.Response.parse_response(binary)
+      {:ok, {:rejected, %CBOR.Tag{tag: :bytes, value: "invalid tx"}}}
+
+      # Disconnected from node
+      iex> binary = Xander.Util.plex_encode(CBOR.encode([3]))
+      iex> Xander.Transaction.Response.parse_response(binary)
+      {:ok, :disconnected}
+
+      # Error handling invalid CBOR
+      iex> Xander.Transaction.Response.parse_response(<<0, 1, 2, 3>>)
+      {:error, :invalid_format}
+
+      # Error handling nil input
+      iex> Xander.Transaction.Response.parse_response(nil)
+      {:error, :invalid_input}
+  """
+  @spec parse_response(binary() | nil) ::
+          {:error,
+           :cannot_decode_non_binary_values
+           | :cbor_function_clause_error
+           | :cbor_match_error
+           | :invalid_format
+           | :invalid_input}
+          | {:ok, :accepted | :disconnected | {:rejected, any()}}
```

Comment:
I feel these variations of success return values are a bit confusing. How about we return one single success return type `{:ok, :accepted}` (we can incorporate returning the tx id at a later stage) and then every other one is an `{:error, some_reason}` ?

Reply:
agreed; this is changed.

Reply:
I agree they were confusing. They were also useful in separating the parsing result from the actual response result, though!

Now, the calling function can't tell a rejected tx from a decoding error - it's now [logging a parse error message](https://github.com/wowica/xander/pull/21/files#diff-9bbbcd154b50aa7f7a1946fe8f5534ce7db56199b1a5b6e65f40035743da0d30R226) when a transaction is rejected.

I'm thinking of moving `{:rejected, reason}` up a level to be alongside `{:ok, :accepted}` and `{:error, _}` in the return type. What do you think @caike ?

Reply:
I like that idea 👍

Reply:
Implemented / RFR again!

Reply:
Beautiful 💯